### PR TITLE
In progress section

### PR DESCRIPTION
**Pull request by DaleDMT**

### CODE REVIEW

File: ckanext/dgua/controllers/in_progress.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    try:
+        organization_id = int(input_data)
+        organization = InProgressOrganization.get_by_id(organization_id)
+        if not organization:
+            raise tk.ValidationError('Not found')
+    except ValueError:
+        h.flash_error(tk._('Invalid input'))
+        tk.redirect_to('in_progress_index')
+    except tk.ValidationError:
+        h.flash_error(tk._('Organization not found'))
```
Comment (Krokop) [inline ckanext/dgua/controllers/in_progress.py]: Should be the same message as on line 17

File: ckanext/dgua/controllers/in_progress.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    try:
+        package_id = int(input_data)
+        package = InProgressPackage.get_by_id(package_id)
+        if not package:
+            raise tk.ValidationError('Not found')
+    except ValueError:
+        h.flash_error(tk._('Invalid input'))
+        tk.redirect_to('in_progress_index')
+    except tk.ValidationError:
+        h.flash_error(tk._('Package not found'))
```
Comment (Krokop) [inline ckanext/dgua/controllers/in_progress.py]: Should be the same message as on line 31

File: ckanext/dgua/templates/in_progress/organization/new.html [hunk 3]
```diff
@@ -0,0 +1,87 @@
+{% extends "in_progress/organization/base.html" %}
+
+{% block primary_content %}
```
Comment (Krokop) [inline ckanext/dgua/templates/in_progress/organization/new.html]: You can create one form, and inherit it in edit.html and new.html.

File: ckanext/dgua/templates/in_progress/package/index.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        <td>{{ item.InProgressOrganization.name }}</td>
+        <td>{% if item.InProgressPackage.url %}<a href="{{ item.InProgressPackage.url }}" target="_blank">{{ item.InProgressPackage.url }}</a>{% endif %}</td>
+        <td>{% if item.InProgressPackage.url %}<i class="fa fa-check" style="color: green"></i>{% endif %}</td>
+        <td>
+            <div class="btn-group pull-right">
+                <a class="btn btn-small" href="{{ h.url_for('in_progress_package_edit', package_id=item.InProgressPackage.id) }}" title="Редагувати">
+                    <i class="fa fa-wrench"></i>
+                </a>
+                <a class="btn btn-danger btn-small" href="{{ h.url_for('in_progress_package_delete', package_id=item.InProgressPackage.id) }}" data-module="confirm-action"
+                   data-module-content="Ви впевнені, що хочете видалити цего распорядника?"
```
Comment (Krokop) [inline ckanext/dgua/templates/in_progress/package/index.html]: {% trans %}

File: ckanext/dgua/templates/in_progress/package/new.html [hunk 5]
```diff
@@ -0,0 +1,60 @@
+{% extends "in_progress/package/base.html" %}
+
+{% block primary_content %}
+    <h1 class="page-heading">{% trans %}New package{% endtrans %}</h1>
+    <div class="thread-post">
+        <form id="thread-form" data-toggle="validator" role="form" method="post">
+
+            <div class="form-group">
+                <label>{% trans %}Package Name{% endtrans %}:</label>
+                <input type="text" placeholder="{% trans %}Фінансова звітність за 2017 рік{% endtrans %}" class="form-control"
```
Comment (Krokop) [inline ckanext/dgua/templates/in_progress/package/new.html]: Фінансова звітність за 2017 рік -> please translate to english.

File: ckanext/dgua/templates/in_progress/package/package_list.html [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            <td>{{ package.name }}</td>
+            <td>{% if package.url %}<a href="{{ package.url }}" target="_blank">{{ package.url }}</a>{% endif %}</td>
+            <td>{% if package.url %}<i class="fa fa-check" style="color: green"></i>{% endif %}</td>
+            <td>
+            <div class="btn-group pull-right">
+              <a class="btn btn-small" href="{{ h.url_for('in_progress_package_edit', package_id=package.id) }}" title="Редагувати">
+                <i class="fa fa-wrench"></i>
+              </a>
+              <a class="btn btn-danger btn-small" href="{{ h.url_for('in_progress_package_delete', organization_id=organization.id, package_id=package.id) }}" data-module="confirm-action"
+                 data-module-content="Ви впевнені, що хочете видалити цего распорядника?"
```
Comment (Krokop) [inline ckanext/dgua/templates/in_progress/package/package_list.html]: {% trans %}

File: ckanext/dgua/templates/in_progress/package/package_list.html [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+                 title="Видалити"><i class="fa fa-times"></i></a>
```
Comment (Krokop) [inline ckanext/dgua/templates/in_progress/package/package_list.html]: {% trans %}

File: ckanext/dgua/controllers/in_progress.py [hunk 8]
```diff
@@ -0,0 +1,146 @@
+from ckan.common import c
+from ckan.lib.base import BaseController, abort
+import ckan.lib.helpers as h
+from ckan.plugins import toolkit as tk
+
+from ckanext.dgua.forms import InProgressOrganizationForm, InProgressPackageForm
+from ckanext.dgua.models import InProgressOrganization, InProgressPackage
+
+from . category import redirect_if_not_sysadmin
```
[the remark this answers is not in the document]
Comment (DaleDMT) [reply ckanext/dgua/controllers/in_progress.py]: Done

File: ckanext/dgua/controllers/in_progress.py (hunk 1 above)
Comment (DaleDMT) [reply ckanext/dgua/controllers/in_progress.py]: Done

File: ckanext/dgua/controllers/in_progress.py (hunk 2 above)
Comment (DaleDMT) [reply ckanext/dgua/controllers/in_progress.py]: Done

File: ckanext/dgua/templates/in_progress/organization/new.html (hunk 3 above)
Comment (DaleDMT) [reply ckanext/dgua/templates/in_progress/organization/new.html]: Done. Thank you, I forgot about refactoring

File: ckanext/dgua/templates/in_progress/package/index.html (hunk 4 above)
Comment (DaleDMT) [reply ckanext/dgua/templates/in_progress/package/index.html]: Done

File: ckanext/dgua/templates/in_progress/package/new.html (hunk 5 above)
Comment (DaleDMT) [reply ckanext/dgua/templates/in_progress/package/new.html]: Done

File: ckanext/dgua/templates/in_progress/package/package_list.html (hunk 6 above)
Comment (DaleDMT) [reply ckanext/dgua/templates/in_progress/package/package_list.html]: Done

File: ckanext/dgua/templates/in_progress/package/package_list.html (hunk 7 above)
Comment (DaleDMT) [reply ckanext/dgua/templates/in_progress/package/package_list.html]: Done

File: ckanext/dgua/controllers/in_progress.py [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+                                    average_qa=form.average_qa.data, is_show=form.is_show.data)
+                h.flash_success(tk._(u'Organization {} successfully changed'.format(organization.name)))
+                tk.redirect_to('in_progress_organization_index')
+
+        form.organization_name.data = organization.name
+        form.metric.data = organization.metric
+        form.average_qa.data = organization.average_qa
+        form.is_show.data = organization.is_show
+
+        return tk.render('in_progress/organization/edit.html', {'form': form, 'organization_list': InProgressOrganization.all(),
```
Comment (Krokop) [inline ckanext/dgua/controllers/in_progress.py]: pep8, max length

File: ckanext/dgua/models.py [hunk 10]
```diff
@@ -468,5 +468,163 @@ def delete(self, commit=True):
...
+        session = Session()
+        session.delete(self)
+        if commit:
+            session.commit()
+
+meta.mapper(InProgressOrganization, in_progress_organization_table)
+
+
+in_progress_package_table = Table('in_progress_package', meta.metadata,
+                                       Column('id', types.Integer, primary_key=True, autoincrement=True),
```
Comment (Krokop) [inline ckanext/dgua/models.py]: pep8

**Pull request merged**